common/persistence/pinot/pinot_visibility_store.go

```diff
@@ -976,6 +1036,51 @@ func getListWorkflowExecutionsQuery(tableName string, request *p.InternalListWor
 	return query.String(), nil
 }
 
+func getListAllWorkflowExecutionsQuery(tableName string, request *p.InternalListAllWorkflowExecutionsByTypeRequest) (string, error) {
+	if request == nil {
+		return "", nil
+	}
+
+	query := NewPinotQuery(tableName)
+
+	query.filters.addEqual(DomainID, request.DomainUUID)
+	query.filters.addEqual(IsDeleted, false)
+
+	if !request.EarliestTime.IsZero() && !request.LatestTime.IsZero() {
+		earliest := request.EarliestTime.UnixMilli() - oneMicroSecondInNano
```

Comment:
The unit here is not matched

Reply:
I actually followed this from the other queries built for pinot. For e.g. https://github.com/uber/cadence/blob/master/common/persistence/pinot/pinot_visibility_store.go#L1025

Reply:
Oh ok, this is adding 1s to endTime and minus 1s to startTime to make it include the earliest/latest time passed in. I will rethink and discuss with Bowen about this implementation.